src/safeposix/syscalls/net_calls.rs

```diff
@@ -739,10 +739,21 @@ impl Cage {
                                 drop(fdtable);
                                 drop(mutmetadata);
 
-                                match sockfdobj.domain {
-                                    PF_INET => sockobj.accept(true),
-                                    PF_INET6 => sockobj.accept(false),
-                                    _ => panic!("Unknown domain in accepting socket"),
+                                match sockfdobj.flags {
+                                    O_NONBLOCK => {
+                                        match sockfdobj.domain {
+                                            PF_INET => sockobj.nonblock_accept(true),
+                                            PF_INET6 => sockobj.nonblock_accept(false),
+                                            _ => panic!("Unknown domain in accepting socket"),
+                                        }
+                                    },
+                                    _ => {
+                                        match sockfdobj.domain {
+                                            PF_INET => sockobj.accept(true),
+                                            PF_INET6 => sockobj.accept(false),
+                                            _ => panic!("Unknown domain in accepting socket"),
+                                        }
+                                    }
```

Comment:
This is incorrect and overkill. Incorrect because flags may contain the O_NONBLOCK bit set but not be equal to it, an if statement with an & check would be what we're looking for.

Reply:
ah, makes sense

Reply:
fixed